lib/app/sequence_base.rb

```diff
@@ -708,10 +710,17 @@ def validate_reference_resolutions(resource, resolved_references = Set.new, max_
                 next
               end
             end
-            value.read
+            reference = value.reference
+            reference_type = value.resource_type
+            resource = value.read
+
+            raise InvalidReferenceResource if resource.resourceType != reference_type
+
             resolved_references.add(value.reference)
           rescue ClientException => e
             problems << "#{path} did not resolve: #{e}"
+          rescue InvalidReferenceResource
```

Comment:
This is an update to the 'platform'.  Is it generic enough to put this here vs. in a test?

Reply:
Yeah, I think verifying that a reference is the correct type is pretty generic functionality.

Reply:
Ok, if you think so I'm good with it.  I just worry because it isn't clear to me when you want to specialize the Exception and when you don't.  If we encourage this pattern then this list of rescues will get mighty long.  Should we have an 'InvalidResource' exception (that seems even more generic than this)?  An 'InvalidFhirUser' exception for SMART?

Reply:
Just looked a little closer and now realize this isn't the main rescue section of sequence_base but is isolated to this section.  That makes more sense!